### ISSUE TITLE
Change all occurences of `T::from_str(s)` to `s.parse::<T>()`

### ISSUE THREAD

**Issue by Kixunil** [rust-bitcoin/rust-bitcoin#3234]
`::from_str` is unidiomatic and produces unhelpful error messages if you're messing with imports.

Please don't do this while we're refactoring the crates.

**Comment by apoelstra** [comment]
We should update CONTRIBUTING.md or whatever our style guide is.

`T::from_str` may be "unidiomatic" but it looks nicer (less line noise) and has a more unique name which is easier to grep for and more descriptive.

But fine by me changing this.

**Comment by Kixunil** [comment]
I like turbofishes. :)

You can `rg 'parse::<T>'` just fine.

**Comment by apoelstra** [comment]
Yes, if you know the type that you're converting to, and the caller used a turbofish instead of type inferenece, that'll work. If not, then you need to grep for `parse.*(` I think to reliably find all calls. Though if you were hoping to find a definition of `parse` you'll be disappointed. You need to grep for `from_str` to find that.

**Comment by Kixunil** [comment]
Thinking about it more I don't think I can come up with a scenario where you'd need to find all instances of `FromStr`/`parse` for an unknown type. If you have a confusing error you likely won't even know it came from one of those in the first place and searching for the string is quicker.

**Comment by apoelstra** [comment]
Right. What you almost always want to do is find the definition -- which is hidden behind a translator function in std that takes you to `from_str`.

Anyway we can change this to the more idiomatic thing in our examples and usage. It's one more thing to remember, but I have to remember it to interact with lots of Rust code, so there's not much cost here.

**Comment by tcharding** [comment]
FWIW many of the `FromStr` usages are because I was not that comfortable with turbo fish when I started hacking on `rust-bitcoin`, and I failed to notice that it saved importing `FromStr` because I was so used to doing so. So big +1 from me to change to use `parse`.

**Comment by Kixunil** [comment]
> I was not that comfortable with turbo fish

Me neither when I learned Rust but it's like Club Mate. One gets used to it. ;)

**Comment by storopoli** [comment]
in issue title: `s/occurences/occurrences/`

**Comment by Kixunil** [comment]
@jamillambert it's the other way around, I meant "please don't spend time on it right now/don't trigger rebases". But I think I can live with it now, since I temporarily gave up on finishing my big PR anyway.

> except examples

Did you do it only to reduce the size of the PR or did you think that examples should stay intact for some reason? Because I definitely think that changing examples might be even more important to not teach people bad habits.

**Comment by jamillambert** [comment]
> Did you do it only to reduce the size of the PR or

I started by using a macro to replace it everywhere but got too many compile errors from special cases that didn't fit the format, so I just focused on one part and went through one at a time.  I can change the examples tomorrow, the majority of cases causing issues were in tests.
